### PR TITLE
DTSEE-5303 - Retries in Playwright Tests

### PR DESCRIPTION
**Pull request by david-gimelle**
https://tools.hmcts.net/jira/browse/DTSSE-5203
https://tools.hmcts.net/jira/browse/DTSSE-5202

Add some retries in seach staff tests
Improving playwright configuration

### CODE REVIEW

**Review by connorpgpmcelroy — DISMISSED** [review]
LGTM

File: playwright.config.ts [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
   forbidOnly: !!process.env.CI,
   /* Retry on CI only */
-  retries: 4, // Set the number of retries for all projects
+  retries: 3, // Set the number of retries for all projects
 
-  timeout: 10 * 60 * 1000,  // 10 minutes
+  timeout: 3 * 60 * 1000,  // 10 minutes
   expect: {
-    timeout: 5 * 60 * 1000,  // 5 minutes
+    timeout: 1 * 60 * 1000,  // 5 minutes
```
Comment (andywilkinshmcts) [inline playwright.config.ts:16]: Remove comment

File: playwright.config.ts [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
   /* Run tests in files in parallel */
   fullyParallel: true,
   /* Fail the build on CI if you accidentally left test.only in the source code. */
   forbidOnly: !!process.env.CI,
   /* Retry on CI only */
-  retries: 4, // Set the number of retries for all projects
+  retries: 3, // Set the number of retries for all projects
 
-  timeout: 10 * 60 * 1000,  // 10 minutes
+  timeout: 3 * 60 * 1000,  // 10 minutes
```
Comment (andywilkinshmcts) [inline playwright.config.ts:14]: Please fix the comment to say 3 minutes - actually remove the comment it's obvious

File: playwright.config.ts [hunk 3]
```diff
@@ -23,39 +25,39 @@ module.exports = defineConfig({
       name: "chromium",
       use: { ...devices["Desktop Chrome"],
         channel: "chrome",
-        headless: false,
+        headless: headless,
```
Comment (andywilkinshmcts) [inline playwright.config.ts:28]: are you sure that you mean headless: headless and not headless: true
headless: "headless" is possible I suppose, but I've very skeptical about what you've got here

File: playwright.config.ts [hunk 4]
```diff
@@ -23,39 +25,39 @@ module.exports = defineConfig({
...
         trace: "on-first-retry",
       },
     },
     {
       name: "firefox",
       use: { ...devices["Desktop Firefox"],
         screenshot: 'only-on-failure',
-        headless: false,
+        headless: headless,
```
Comment (andywilkinshmcts) [inline playwright.config.ts:36]: fix

File: playwright.config.ts [hunk 5]
```diff
@@ -23,39 +25,39 @@ module.exports = defineConfig({
...
         trace: 'off'
       },
     },
     {
       name: "webkit",
       use: { ...devices["Desktop Safari"],
         screenshot: 'only-on-failure',
-        headless: false,
+        headless: headless,
```
Comment (andywilkinshmcts) [inline playwright.config.ts:44]: fix

File: playwright.config.ts [hunk 6]
```diff
@@ -23,39 +25,39 @@ module.exports = defineConfig({
...
         trace: 'off'
       },
     },
     {
       name: "MobileChrome",
       use: { ...devices["Pixel 5"],
         screenshot: 'only-on-failure',
-        headless: false,
+        headless: headless,
```
Comment (andywilkinshmcts) [inline playwright.config.ts:52]: fix

File: playwright.config.ts [hunk 7]
```diff
@@ -23,39 +25,39 @@ module.exports = defineConfig({
...
         trace: 'off'
       },
     },
     {
       name: "MobileSafari",
       use: { ...devices["iPhone 12"],
         screenshot: 'only-on-failure',
-        headless: false,
+        headless: headless,
```
Comment (andywilkinshmcts) [inline playwright.config.ts:60]: fix

File: playwright.config.ts [hunk 8]
```diff
@@ -64,7 +66,7 @@ module.exports = defineConfig({
       use: { ...devices["Desktop Edge"],
         channel: "msedge",
         screenshot: 'only-on-failure',
-        headless: false,
+        headless: headless,
```
Comment (andywilkinshmcts) [inline playwright.config.ts:69]: fix

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]

File: playwright.config.ts (hunk 1 above)
Comment (david-gimelle) [reply playwright.config.ts]: done

File: playwright.config.ts (hunk 2 above)
Comment (david-gimelle) [reply playwright.config.ts]: done

File: playwright.config.ts (hunk 3 above)
Comment (david-gimelle) [reply playwright.config.ts]: I have changed this variable to headlessMode. The default value is true.

File: playwright.config.ts (hunk 4 above)
Comment (david-gimelle) [reply playwright.config.ts]: done

File: playwright.config.ts (hunk 6 above)
Comment (david-gimelle) [reply playwright.config.ts]: done

File: playwright.config.ts (hunk 8 above)
Comment (david-gimelle) [reply playwright.config.ts]: done

File: playwright_tests/E2E/steps/staff-steps.ts [hunk 9]
```diff
@@ -1,15 +1,17 @@
...
+
+export async function clickToStaffPage(page) {
+  console.log("Going to Staff Page");
+  await retryAction(async () => {
+    await page.getByRole('link', { name: 'Staff' }).click();
+    await page.waitForSelector('text=User list', { state: 'visible', timeout: 1000 });
+  });
+}
+
+export async function fillSearchBox(page) {
```
Comment (andywilkinshmcts) [inline playwright_tests/E2E/steps/staff-steps.ts:11]: I'd be happier if this function had an argument for the text to submit, perhaps with a default, e.g. function fillSearchBox(page, text='xui').

**Review by andywilkinshmcts — APPROVED** [review]

**Pull request merged**